From c036138c3b2164890eba45b6d07db5c9110f9839 Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 12 Oct 2018 15:40:22 +0530
Subject: [PATCH 0471/1851] v4l: xilinx-multi-scaler: Remove volatile

commit 31ff947476b863c4f59eb26baa654631e7b05a7b from
https://github.com/Xilinx/linux-xlnx.git

As per Documentation/process/volatile-considered-harmful.rst,
using accessor functions do not require volatile as accessors
are written to prevent unwanted optimization.

And we are using ioread32 and iowrite32, so volatile is not require.

This patch removes the volatile from code and remove
the below warning also

warning: passing argument 1 of 'ioread32' discards 'const volatile'
qualifier from pointer target type [-Wdiscarded-qualifiers]
     return ioread32(addr);
                     ^~~~

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 45cf6f822ee7..7654f400b9d4 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -356,18 +356,18 @@ struct xm2m_msc_dev {
 
 #define fh_to_chanctx(__fh) container_of(__fh, struct xm2msc_chan_ctx, fh)
 
-static inline u32 xm2msc_readreg(const volatile void __iomem *addr)
+static inline u32 xm2msc_readreg(const void __iomem *addr)
 {
 	return ioread32(addr);
 }
 
-static inline void xm2msc_write64reg(volatile void __iomem *addr, u64 value)
+static inline void xm2msc_write64reg(void __iomem *addr, u64 value)
 {
 	iowrite32(lower_32_bits(value), addr);
 	iowrite32(upper_32_bits(value), (void __iomem *)(addr + 4));
 }
 
-static inline void xm2msc_writereg(volatile void __iomem *addr, u32 value)
+static inline void xm2msc_writereg(void __iomem *addr, u32 value)
 {
 	iowrite32(value, addr);
 }
@@ -647,7 +647,7 @@ xm2msc_pr_chanctx(struct xm2msc_chan_ctx *ctx, const char *fun_name)
 }
 
 static void
-xm2msc_pr_screg(struct device *dev, const volatile void __iomem *base)
+xm2msc_pr_screg(struct device *dev, const void __iomem *base)
 {
 	dev_dbg(dev, "Ctr, GIE,  IE,  IS   OUT\n");
 	dev_dbg(dev, "0x%x  0x%x   0x%x  0x%x  0x%x\n",
-- 
2.31.1

